### PR TITLE
Add issue link info to PR template

### PR DESCRIPTION
**Pull request by stmcginnis**
While reviewing pull requests I noticed there are a lot of PRs that are addressing open issues, but the PRs are not properly linked. This results in a lot of issues being left open, even after the fix has been merged.

This updates the PR template to give a short instruction on adding the expected line in the PR description so that the issue and PR can be linked automatically and closed on merge.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -22,3 +22,15 @@
  https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.
 
 -->
+
+<!--
+If this pull request resolves an open issue, please link the issue in the PR
+description so it will automically close when the PR is merged. This is done
```
Comment (sftim) [inline .github/PULL_REQUEST_TEMPLATE.md:28]:
```suggestion
description so it will automatically close when the PR is merged. This is done
```

**Comment by Shubham82** [comment]
Thanks @stmcginnis 
/lgtm

**Review by Gauravpadam — COMMENTED** [review]
Thanks @stmcginnis,

Do you feel we should keep this somewhere near the beginning of the file?
Also, Should keep it short, In a single line perhaps?
"If this PR solves an open issue, please add a closing tag (Closes #[issue_number]) to the PR description."

**Comment by stmcginnis** [comment]
That makes sense. I've pushed up another format that has worked really well for me on other projects. It makes it more clear what sections/information to fill out in the PR template.

It does not shorten things as you mention, but I wanted to throw it out there as another potential approach. If you prefer the original way but more concise, just let me know and I can switch things back to that.

File: pmsecret.yaml
Comment (sftim) [inline pmsecret.yaml:1]: What's this?

**Review by sftim — CHANGES_REQUESTED** [review]
This PR doesn't look right.

File: pmsecret.yaml
Comment (stmcginnis) [reply pmsecret.yaml:1]: Oops, accidental `git add .` there. Cleaned up now.

**Comment by aj11anuj** [comment]
I liked the concise format Gaurav proposed -
> "If this PR solves an open issue, please add a closing tag (Closes #[issue_number]) to the PR description."

and if needed then a hyperlink for newcomers to learn more about it can be added at bottom of this whole description.

**Comment by Shubham82** [comment]
@tengqm, could you please take a look?

**Pull request merged**